pkg/controller/nmstate/nmstate_controller_test.go

```diff
@@ -109,24 +109,32 @@ var _ = Describe("NMState controller reconcile", func() {
 				&nmstatev1beta1.NMState{},
 			)
 			// set NodeSelector field in operator Spec
-			nmstate.Spec.NodeSelector = dsNodeSelector
+			nmstate.Spec.NodeSelector = handlerNodeSelector
 			objs := []runtime.Object{&nmstate}
 			// Create a fake client to mock API calls.
 			cl = fake.NewFakeClientWithScheme(s, objs...)
 			reconciler.client = cl
 			request.Name = existingNMStateName
-		})
-		It("should add NodeSelector labels to Daemonsets spec", func() {
 			result, err := reconciler.Reconcile(request)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(result).To(Equal(reconcile.Result{}))
-			// verify NodeSelector value for both worker and master DS
+		})
+		It("should add NodeSelector to handler daemonset", func() {
 			ds := &appsv1.DaemonSet{}
-			for _, c := range []string{"worker", "master"} {
-				cl.Get(context.TODO(), types.NamespacedName{Namespace: handlerNamespace, Name: handlerPrefix + "-nmstate-handler-" + c}, ds)
-				for k, v := range dsNodeSelector {
-					Expect(ds.Spec.Template.Spec.NodeSelector).To(HaveKeyWithValue(k, v))
-				}
+			handlerKey := types.NamespacedName{Namespace: handlerNamespace, Name: handlerPrefix + "-nmstate-handler"}
+			err := cl.Get(context.TODO(), handlerKey, ds)
+			Expect(err).ToNot(HaveOccurred())
+			for k, v := range handlerNodeSelector {
+				Expect(ds.Spec.Template.Spec.NodeSelector).To(HaveKeyWithValue(k, v))
+			}
+		})
+		It("should NOT add NodeSelector to webhook deployment", func() {
```

Comment:
Any special reason to add this test?  you didn't include the  '.HandlerNodeSelector' in webhook deployment manifest?

Reply:
The operator's CR NodeSelector is only for handlers (we want to skip nodes that does not support nmstate) webhook is transparent for that since is just ensuring the nncp status is correctly reset.